### PR TITLE
ci: add copy files and create pr action

### PR DESCRIPTION
**Pull request by yangchiu**
ci: add copy files and create pr action

For https://github.com/longhorn/longhorn/issues/5854

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

File: lib/copy-files-and-create-pr-action/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+git add .
+if git status | grep -q "Changes to be committed"; then
+  git commit --message "${INPUT_COMMIT_MSG}"
+  echo "Pushing git commit"
+  git push -u origin "HEAD:${INPUT_DESTINATION_HEAD_BRANCH}"
+  echo "Creating a pull request"
+  gh pr create -t "${INPUT_PR_TITLE}" \
+               -b "${INPUT_COMMIT_MSG}" \
+               -B "${INPUT_DESTINATION_BASE_BRANCH}" \
+               -H "${INPUT_DESTINATION_HEAD_BRANCH}" \
```
Comment (innobead) [inline lib/copy-files-and-create-pr-action/entrypoint.sh:41]: `INPUT_DESTINATION_HEAD_BRANCH` seems only able to delete when the PR merged, right?

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. 

Besides master/main, can we support custom branches by using glob pattern by a parameter?

File: lib/copy-files-and-create-pr-action/entrypoint.sh (hunk 1 above)
Comment (yangchiu) [reply lib/copy-files-and-create-pr-action/entrypoint.sh:41]: Yes, if you delete it first, the corresponding PR will be closed too.

**Comment by yangchiu** [comment]
> In general, LGTM.
> 
> Besides master/main, can we support custom branches by using glob pattern by a parameter?

Modified in https://github.com/longhorn/longhorn-manager/pull/1913#issuecomment-1566508593

**Comment by yangchiu** [comment]
@innobead This PR isn't merged yet.

**Review by innobead — APPROVED** [review]
Ah it was missed 😅

**Pull request merged**